File: day7/part2_bruteforce.py
crabs = [*map(int, open('input.txt').readline().split(','))]

fuel_cost = lambda distance: (1 + distance) * distance // 2

print(min(
	sum(fuel_cost(abs(crab - x)) for crab in crabs)
	for x in range(min(crabs), max(crabs) + 1)))     # Try every x position

File: day7/part2_bruteforce.rb
def fuel_cost(distance); (1 + distance) * distance / 2 end

crabs = File.read('input.txt').split(',').map(&:to_i)

puts (crabs.min..crabs.max).map { |x|
	crabs.sum { |crab_x| fuel_cost (x - crab_x).abs }
}.min